app/services/threat_level/assessment_run_status_data_builder_service.rb

```diff
@@ -0,0 +1,97 @@
+module ThreatLevel
+  class AssessmentRunStatusDataBuilderService
+    def self.get(run)
+      AssessmentRunStatusDataBuilderService.new(run).get
+    end
+
+    def initialize(run)
+      @run = run
+      @newest_result = results.sort_by {|r| r.created_at }.last
+    end
+
+    def get
+      {
+        start_date: @run.start_date
+      }.merge status_and_explanation
+    end
+    private
+    def status_and_explanation
+      case
+      when finished?
+        finished_status_and_explanation
+      when started?
+        started_status_and_explanation
+      when running?
+        running_status_and_explanation
+      when warning?
+        warning_status_and_explanation
+      else
+        error_status_and_explanation
+      end
+    end
+
+    def finished?
+      @run.status == 'finished'
+    end
+
+    def finished_status_and_explanation
+      {
+        status: :finished,
+        explanation: 'Assessment run finished'
+      }
+    end
+
+    def started?
+      @run.start_date > (Time.now - 1.hour) && @newest_result.nil?
```

Comment:
Status is started if run was started less than 1 hour ago and it has not produced any results yet. I assumed 1 hour is long enough to produce results for threat assessment workflow.